src/OrderFormAcceptanceTests.Actions/Pages/OrderForm.cs

```diff
@@ -750,7 +756,19 @@ public void EnterPriceInputValue(string value)
 
         public bool OrderUnitIsDisplayed()
         {
-            return Driver.FindElements(Objects.Pages.OrderForm.OrderUnit).Count == 1;
+            const string pattern = "per (.*)( per (month|year)){0,1}";
+
+            var unitOfPriceLabels = Driver.FindElements(Objects.Pages.OrderForm.OrderUnit).Select(e => e.Text);
+
+            foreach (var label in unitOfPriceLabels)
+            {
+                if (Regex.Match(label, pattern).Success)
+                {
+                    return true;
+                }
+            }
+
+            return false;
```

Comment:
Could this be written as a LINQ statement? This is from memory so could well be wrong but perhaps something like

```c#
unitOfPriceLabels.Any(l => Regex.Match(l, pattern).Success)
```

I'm fairly open as far LINQ vs imperative goes but I think the LINQ statement is a bit more readable here (assuming it's correct).

Also, just wanted to double-check that you only need one item to match the regex here?

Reply:
There are 2 labels that match the By in FindElements, one is the label above the input, one is the label that displays the pricing unit. The check is to make sure that the pricing unit is displayed (obviously doesn't care what the pricing unit actually is, but needs to be displayed on the page)

I've added the Linq statement in the next commit